public/app/plugins/datasource/elasticsearch/dataquery.gen.ts

```diff
@@ -0,0 +1,368 @@
+  settings?: {
+    missing?: string;
+  };
+}
+
+export type InlineScript = (string | {
+    inline?: string
+  });
+
+export interface MetricAggregationWithInlineScript extends BaseMetricAggregation {
+  settings?: {
+    script?: InlineScript;
+  };
+}
+
+export interface Count extends BaseMetricAggregation {
+  type: 'count';
+}
+
+export interface Average extends MetricAggregationWithField, MetricAggregationWithMissingSupport, MetricAggregationWithInlineScript {
+  field?: string;
+  settings?: {
+    script?: (InlineScript | InlineScript);
+    missing?: string;
+  };
+  type: 'avg';
+}
+
+export interface Sum extends MetricAggregationWithField, MetricAggregationWithInlineScript {
+  field?: string;
+  settings?: {
+    script?: (InlineScript | InlineScript);
+    missing?: string;
+  };
+  type: 'sum';
+}
+
+export interface Max extends MetricAggregationWithField, MetricAggregationWithInlineScript {
+  field?: string;
+  settings?: {
+    script?: (InlineScript | InlineScript);
+    missing?: string;
+  };
+  type: 'max';
+}
+
+export interface Min extends MetricAggregationWithField, MetricAggregationWithInlineScript {
+  field?: string;
+  settings?: {
+    script?: (InlineScript | InlineScript);
+    missing?: string;
+  };
+  type: 'max';
+}
+
+export type ExtendedStatMetaType = ('avg' | 'min' | 'max' | 'sum' | 'count' | 'std_deviation' | 'std_deviation_bounds_upper' | 'std_deviation_bounds_lower');
+
+export interface ExtendedStat {
+  label: string;
+  value: ExtendedStatMetaType;
+}
+
+export interface ExtendedStats extends MetricAggregationWithField, MetricAggregationWithInlineScript {
+  field?: string;
+  meta?: Record<string, unknown>;
+  settings?: {
+    script?: (InlineScript | InlineScript);
+    missing?: string;
+    sigma?: string;
+  };
+  type: 'extended_stats';
+}
+
+export interface Percentiles extends MetricAggregationWithField, MetricAggregationWithInlineScript {
+  field?: string;
+  settings?: {
+    script?: (InlineScript | InlineScript);
+    missing?: string;
+    percents?: Array<string>;
+  };
+  type: 'percentiles';
+}
+
+export interface UniqueCount extends MetricAggregationWithField {
+  settings?: {
+    precision_threshold?: string;
+    missing?: string;
+  };
+  type: 'cardinality';
+}
+
+export interface RawDocument extends BaseMetricAggregation {
+  settings?: {
+    size?: string;
+  };
+  type: 'raw_document';
+}
+
+export interface RawData extends BaseMetricAggregation {
+  settings?: {
+    size?: string;
+  };
+  type: 'raw_data';
+}
+
+export interface Logs extends BaseMetricAggregation {
+  settings?: {
+    limit?: string;
+  };
+  type: 'logs';
+}
+
+export interface Rate extends MetricAggregationWithField {
+  settings?: {
+    unit?: string;
+    mode?: string;
+  };
+  type: 'rate';
+}
+
+export interface BasePipelineMetricAggregation extends MetricAggregationWithField {
+  pipelineAgg?: string;
+  type: string;
+}
```

Comment:
@Clarity-89 What about generated type for `BasePipelineMetricAggregation` that produce invalid/incorrect types: 

```
Interface 'BasePipelineMetricAggregation' incorrectly extends interface 'MetricAggregationWithField'.
  Types of property 'type' are incompatible.
    Type 'string' is not assignable to type 'MetricAggregationType'.
```

Not sure why this is happening and where `type: string` is coming from, because it should be `type: PipelineMetricAggregationType`. And even if it was over-ruled in `BasePipelineMetricAggregation` and then  by `BaseMetricAggregation` that again extends it, the type should be `type: MetricAggregationType;` where `MetricAggregationType` can be `PipelineMetricAggregationType`. I am really confused where does the `string` type is coming from.

Reply:
I tried debugging this yesterday but still don't understand where the string type comes from. Looks like cue incorrectly resolves some nested type extensions. I'll create an issue for that and meanwhile, as a workaround, we'll need to extend `BasePipelineMetricAggregation` manually. First we comment out the `type` field in both `#BasePipelineMetricAggregation` and `#PipelineMetricAggregationWithMultipleBucketPaths`, maybe also add a TODO to the comment, so we know it needs to be fixed. Then we manually extend them with the correct type in `types.ts`:
```
// types.ts
import {
  PipelineMetricAggregationType,
  BasePipelineMetricAggregation as SchemaBasePipelineMetricAggregation,
  PipelineMetricAggregationWithMultipleBucketPaths as SchemaPipelineMetricAggregationWithMultipleBucketPaths,
} from './dataquery.gen';

export interface BasePipelineMetricAggregation extends SchemaBasePipelineMetricAggregation {
  type: PipelineMetricAggregationType;
}

export interface PipelineMetricAggregationWithMultipleBucketPaths
  extends SchemaPipelineMetricAggregationWithMultipleBucketPaths {
  type: PipelineMetricAggregationType;
}
```
All the other types will import these types from `type.ts`, and I recommend re-exporting all the types from `./dataquery.gen` inside `type.ts` so it becomes the single source of the types. I see that the generated types that extend these two base types define their own `type` field, so this shouldn't be an issue.